### PR TITLE
Unique Interactsh URL

### LINKED ISSUE #1068: Unique payload per Interactsh placeholder
**Is your feature request related to a problem? Please describe.**

Same **interactsh** payload is generated by a template that contains multiple `{{interactsh-url}}` placeholders within the same template; therefore, once we receive interaction, it is impossible to determine which payload, header, or parameter was used because all of the interactsh payloads are the same.

Example template:  [oob-param-based-interaction](https://github.com/projectdiscovery/nuclei-templates/blob/e9f81943b64a66b5823555249cde28a26af2edcc/vulnerabilities/generic/oob-param-based-interaction.yaml)

**Describe the solution you'd like**

Creating a unique payload for each occurrence of the `{{interactsh-url}}` placeholder.

### PR DESCRIPTION
**Pull request by parrasajad**
## Proposed changes

closes #1068 

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: v2/pkg/protocols/common/interactsh/interactsh.go [hunk 1]
```diff
@@ -227,30 +229,33 @@ type RequestData struct {
...
+			// If we have previous interactions, get them and process them.
+			interactions, ok := interaction.Value().([]*server.Interaction)
+			if !ok {
+				c.requests.Set(id, data, c.eviction)
+				return
 			}
+			matched := false
+			for _, interaction := range interactions {
+				if c.processInteractionForRequest(interaction, data) {
+					matched = true
```
Comment (forgedhallpass) [inline v2/pkg/protocols/common/interactsh/interactsh.go:247]: why not do the `c.interactions.Delete(id)` simply here?

**Review by forgedhallpass — CHANGES_REQUESTED** [review]
Please write at least one unit test that validates the changes.

**Review by ehsandeep — APPROVED** [review]
Tests are passing locally

**Review by Ice3man543 — APPROVED** [review]
lgtm!

**Pull request merged**